### ISSUE TITLE
Is it possible to disable shared chat? 

### ISSUE THREAD

**Issue by CristianGrecu** [fgl27/SmartTwitchTV#241]
Hi, I was wondering if it would be possible to disable shared chat. I was watching The Game Awards and 3 streamers that co-host decided to use the new shared chat feature. It was very chaotic because I usually watch the one with the least amount of viewers.

I know that the moderation becomes shared too, so it might not be easy to do.

**Comment by fgl27** [comment]
I'm not sure if the information is there to filter out, as the app only connects to the chat of the current stream. If something else is coming, it is because it is sent to that stream chat.

I'll check.

**Comment by CristianGrecu** [comment]
That's what I was thinking, it seems pretty baked in, all the mods get access to all the chats etc.

**Comment by fgl27** [comment]
I believe is possible to filter out msg not from the stream, but to test this I need a stream that is receiving the shared chat.

I can find one, if anyone has one please share the stream here when it is live if I'm available I will check.

**Comment by fgl27** [comment]
maybe not sure let me check

**Comment by fgl27** [comment]
no the api for the vod doesn't return chat origing

**Comment by fgl27** [comment]
maybe they updated the vod to do so, but now it doesn't... but the live may have the info need a live that is receiving chats from a different chant to test

**Comment by CristianGrecu** [comment]
Is there a way to see if shared chat is enabled (on PC) and I don't see it? I knew with that vod because I recognize the names and I know it's 3 chats in 1.

**Comment by CristianGrecu** [comment]
I see, thanks. I'll keep an eye out and if I find one live I'll post it here.

**Comment by CristianGrecu** [comment]
So I found some streams live right now with shared chat, and I found a bug as well. 

First the streams : Xaryu - Anniefuchsia - Grubby 

And the bug that I found: if only one stream has shared chat enabled it seems to break multistream on my TV. So 2 streams in the middle with chats on both sides works fine. If only one stream has shared chat it seems to break everything leading to error 2000.  So if I take a random stream from my list and add Grubby's stream right now the stream added on top breaks. If I select 2 other streams without shared chat it works perfectly.

**Comment by fgl27** [comment]
OK,. I wasn't online to see.
Week keep trying.

**Comment by CristianGrecu** [comment]
Yep, it's not a big deal, I mostly wanted to tell you about the "bug" which was my mistake.

Btw, I don't know how many people use multistream often, but If you want a quick fix you could add an option to turn off one of the chats (if it's a duplicate). It shouldn't mess up the aspect ratio that much. Like have 2 streams on top of each other and maybe switch the chat off along with the audio.

**Comment by fgl27** [comment]
I can look into that option once we manage this shared thing...

I made this app to solve the problem. It is hardcoded to filter out, so if it works, you can't disable it.

try and let me know

[app-debug.zip](https://github.com/user-attachments/files/18203346/app-debug.zip)

This is a zip file. Unzip it to install the APK. The app with a yellow icon is the test one.

**Comment by CristianGrecu** [comment]
Thank you! Will check as soon as I find some more streams.

**Comment by CristianGrecu** [comment]
Ok, I tested it, it doesn't seem to do anything.

I used Sodapoppin and Vei, they're live right now doing the co-stream. I checked with multistream and in both apps the chats are 100% identical.

**Comment by fgl27** [comment]
I see, I managed to get some chats

the  "room-id": and "source-room-id" don't match when it come from a diff

`
{
    "raw": "@badge-info=;badges=arcane-season-2-premiere/1;color=#1E90FF;display-name=iGeevs;emotes=;flags=;id=47f11900-9b4d-40bd-94fe-0846323efe79;mod=0;room-id=26301881;source-badge-info=subscriber/25;source-badges=subscriber/24,arcane-season-2-premiere/1;source-id=b1dc27ed-2ba8-4f92-bcfa-8d027ea0851d;source-room-id=97245742;subscriber=0;tmi-sent-ts=1734723183744;turbo=0;user-id=58605027;user-type= :igeevs!igeevs@igeevs.tmi.twitch.tv PRIVMSG #sodapoppin :Fishinge bait successful",
    "tags": {
        "badge-info": true,
        "badges": "arcane-season-2-premiere/1",
        "color": "#1E90FF",
        "display-name": "iGeevs",
        "emotes": true,
        "flags": true,
        "id": "47f11900-9b4d-40bd-94fe-0846323efe79",
        "mod": "0",
        "room-id": "26301881",
        "source-badge-info": "subscriber/25",
        "source-badges": "subscriber/24,arcane-season-2-premiere/1",
        "source-id": "b1dc27ed-2ba8-4f92-bcfa-8d027ea0851d",
        "source-room-id": "97245742",
        "subscriber": "0",
        "tmi-sent-ts": "1734723183744",
        "turbo": "0",
        "user-id": "58605027",
        "user-type": true
    },
    "prefix": "igeevs!igeevs@igeevs.tmi.twitch.tv",
    "command": "PRIVMSG",
    "params": [
        "#sodapoppin",
        "Fishinge bait successful"
    ]
}
`

`
{
    "raw": "@badge-info=;badges=arcane-season-2-premiere/1;client-nonce=a7b38b7c6e6acf6a0de380506aa04982;color=#1E90FF;display-name=oXzynn;emotes=;first-msg=0;flags=;id=686c2635-bea6-4028-9d7c-ecc3b769931c;mod=0;returning-chatter=0;room-id=26301881;source-badge-info=;source-badges=arcane-season-2-premiere/1;source-id=686c2635-bea6-4028-9d7c-ecc3b769931c;source-room-id=26301881;subscriber=0;tmi-sent-ts=1734723183704;turbo=0;user-id=594118754;user-type= :oxzynn!oxzynn@oxzynn.tmi.twitch.tv PRIVMSG #sodapoppin :I miss omegareviewer Sadge",
    "tags": {
        "badge-info": true,
        "badges": "arcane-season-2-premiere/1",
        "client-nonce": "a7b38b7c6e6acf6a0de380506aa04982",
        "color": "#1E90FF",
        "display-name": "oXzynn",
        "emotes": true,
        "first-msg": "0",
        "flags": true,
        "id": "686c2635-bea6-4028-9d7c-ecc3b769931c",
        "mod": "0",
        "returning-chatter": "0",
        "room-id": "26301881",
        "source-badge-info": true,
        "source-badges": "arcane-season-2-premiere/1",
        "source-id": "686c2635-bea6-4028-9d7c-ecc3b769931c",
        "source-room-id": "26301881",
        "subscriber": "0",
        "tmi-sent-ts": "1734723183704",
        "turbo": "0",
        "user-id": "594118754",
        "user-type": true
    },
    "prefix": "oxzynn!oxzynn@oxzynn.tmi.twitch.tv",
    "command": "PRIVMSG",
    "params": [
        "#sodapoppin",
        "I miss omegareviewer Sadge"
    ]
}

`

**Comment by fgl27** [comment]
I will share an app to filter this out later

**Comment by CristianGrecu** [comment]
No worries, glad you caught it live.

**Comment by fgl27** [comment]
Hey, I have added full shared chat support.

![image](https://github.com/user-attachments/assets/f3233023-4ad6-49dd-815a-85059912b719)


Please test and let me know.

**Comment by CristianGrecu** [comment]
Yep, it seems to work perfectly. Thank you!!

**Comment by lextra2** [comment]
> Please test and let me know.

Disabling doesn't work.

**Comment by CristianGrecu** [comment]
Yesterday I made sure it worked before posting, but today it seems that something bugged out.  Someone else mentioned they can't disable it, I can't seem to be able to enable it now.

When I checked it yesterday I used multistream to see if they match > disabled it in the settings and it worked fine. Today even with the setting set to "no" I only get the chat from the stream that I am watching. 

Any ideas?

**Comment by CristianGrecu** [comment]
I don't know if it helps, but one of the gifted subs alerts came through from the other stream, regular messages don't.

**Comment by CristianGrecu** [comment]
https://github.com/user-attachments/assets/495d2d97-885b-442a-a786-cd7ad3d32ab9


Here's an example, shared chat "disabled" but it is actually on with missing badges.

**Comment by fgl27** [comment]
Is fine for me you can test using channel shared_chat_test_01 02 or 03

Wolfsbanee is offline with shared off

Was the sub chat orange color sub message

Or different color msg?

**Comment by CristianGrecu** [comment]
Maybe there was something wrong with those 5 streams? I tested on all my devices and the settings are reverse everywhere.

So, as you can see in the second clip, with it "enabled" it's actually off and the only thing that got through from another chat was a message showing 10 gifted subs.

You can see that in the first clip the shared chat is supposed to be off but it's actually on with broken badges. I would post longer videos but there's that file limit...

**Comment by CristianGrecu** [comment]
I just tested it right now with this stream: "Ziqoftw" . Same exact thing, the settings are swapped and the badges are missing when shared chat is working. Just like in the second clip, all of the people from the other streams have black gaps next to their names.

**Comment by fgl27** [comment]
The setting option says 'disabled shared chat'

![{A32AD518-4B2A-4307-B8C3-7CB1B1DE03C1}](https://github.com/user-attachments/assets/b31819a7-e661-401d-8fd9-34201081f5b2)

if you set it to yes it means you disabled it, and no shared is enabled

the settings are not 'enabled shared', get it?

looks fine to me, other than on the first video the badges didn't load, sometimes takes a few seconds for it to load.

**Comment by CristianGrecu** [comment]
Yes, of course I know, but look at the videos from above. In the first chat it is clearly "YES" which means shared chat should be off, but you can see the chats sync perfectly. In the second video it is clearly "NO" which should mean it is on but it is not..

**Comment by CristianGrecu** [comment]
I'm sorry if I'm bad at explaining, what I am trying to say is that it's an illusion, don't be fooled by the middle prompt showing the list of streams because the actual chats aren't merged. I tested 10 different times today, I'm telling you that the chats don't match what the app is saying.

**Comment by fgl27** [comment]
No is all fine... is just the videos are difficult to understand...

I pushed an update seems OK now, let me know if not

**Comment by fgl27** [comment]
BTW you can disable shared chat from the player

![image](https://github.com/user-attachments/assets/48cac1d6-7597-473e-be50-ad01fcdb620d)


![image](https://github.com/user-attachments/assets/2271f8fe-5301-48c8-b6af-ca3f5e1a6b58)

**Comment by CristianGrecu** [comment]
What do you mean difficult to understand? I was watching wolfsbanee with the setting set correctly and no other chat but hers was showing...

**Comment by fgl27** [comment]
is difficult you may understand, but it is confusing all happens too fast

**Comment by CristianGrecu** [comment]
Well now it seems to work perfectly, so I don't know what you mean. In the videos above it was clearly bugged, now it seems to be fixed. I was never confused about how it's supposed to work.

**Comment by CristianGrecu** [comment]
I waited 10 minutes with the setting set to "no" watching wolfsbanee and never saw a different badge. Now watching ziqoftw it works fine.

**Comment by CristianGrecu** [comment]
Also, I am comparing my tablet (without the update) with my tv (with update). My tablet's app is still broken like in the videos above.

I'm kind of dissapointed that you would think I'm dumb or confused when I clearly showed proof it was broken. Oh well.

**Comment by fgl27** [comment]
I can not get confused.

**Comment by fgl27** [comment]
I'm a magic person that can confuse a thing?

**Comment by CristianGrecu** [comment]
No, I was just hoping that we could agree that in my first videos that shared chat is disabled, but you can see in multi-stream how they match, you can literally draw a line from one person to the other and they are from all the streams. That's all I was saying. 

In the second video the chats don't even advance with shared turned on, they should be super fast like in the first.

If I got it wrong somehow I apologize.

**Comment by fgl27** [comment]
The video is hard to understand, at least for me, if not for you 

you must understand that different people see things differently some people have all kinds of issues to see things It doesn't matter that two people see the same thing they may understand the same differently

is a fact of how people work :)

**Comment by CristianGrecu** [comment]
If you look at the chatter midnight_serpent he uses a HUH emote which is a cat in the left stream but the streamers face in the second. If shared chat was off (it shows clearly in the settings disable shared chat : YES)

How would midnight serpent be able to comment on both chats at the same time if it wasn't a shared chat?

**Comment by CristianGrecu** [comment]
I didn't mean to be rude, I'm sorry. It's just that I left the settings in the video on purpose so you can see that I am not lying. With it on YES, there shouldn't have been this chatter for example midnight serpent on both left and right, it is not possible without sharing the chat.

**Comment by CristianGrecu** [comment]
Also another big evidence, the streamlabs message showing cooksux, which is a streamer I wasn't even watching in the video. Which shows again that shared chat went through even with the setting on "YES" (yes - disable).

**Comment by fgl27** [comment]
Is all good... some devices may take a few minutes to update if one of your devices is not yet aware a few minutes close the app and reopen it to see if it did

**Comment by CristianGrecu** [comment]
Well that's the thing, with the setting set to "NO" so disable shared - NO , I waited 10 mins watching wolfsbanee and I swear I only saw her badge and like 7 messages, not a single message from the other 4 streams. That's why I said it was bugged.

**Comment by CristianGrecu** [comment]
It might've been a glitch or something. Again I'm sorry if I came off as rude, maybe I should've tried to explain better.
